remote-method/index.js

```diff
@@ -0,0 +1,244 @@
+'use strict';
+var yeoman = require('yeoman-generator');
+var chalk = require('chalk');
+
+var workspace = require('loopback-workspace');
+var ModelProperty = workspace.models.ModelProperty;
+
+var actions = require('../lib/actions');
+var helpers = require('../lib/helpers');
+var validateName = helpers.validateName;
+var accepts = [];
+var returns = [];
+
+module.exports = yeoman.generators.Base.extend({
+  // NOTE(bajtos)
+  // This generator does not track file changes via yeoman,
+  // as loopback-workspace is editing (modifying) files when
+  // saving project changes.
+
+  help: function() {
+    return helpers.customHelp(this);
+  },
+
+  loadProject: actions.loadProject,
+
+  loadModels: actions.loadModels,
+
+  askForModel: function() {
+    if (this.options.modelName) {
+      this.modelName = this.options.modelName;
+      return;
+    }
+
+    var done = this.async();
+    var prompts = [
+      {
+        name: 'model',
+        message: 'Select the model:',
+        type: 'list',
+        choices: this.modelNames
+      }
+    ];
+
+    this.prompt(prompts, function(answers) {
+      this.modelName = answers.model;
+      done();
+    }.bind(this));
+  },
+
+  findModelDefinition: function() {
+    this.modelDefinition = this.projectModels.filter(function(m) {
+      return m.name === this.modelName;
+    }.bind(this))[0];
+
+    if (!this.modelDefinition) {
+      var msg = 'Model not found: ' + this.modelName;
+      this.log(chalk.red(msg));
+      this.async()(new Error(msg));
+    }
+  },
+
+  askForParameters: function() {
+    var done = this.async();
+
+    var prompts = [
+      {
+        name: 'methodName',
+        message: 'Enter the remote method name:',
+        required: true,
+        validate: function(value) {
+          return validateName(value);
+        }
+      },
+      {
+        name: 'isStatic',
+        message: 'Is Static?',
+        required:true,
+        type: 'confirm',
+        default: true
+      },
+      {
+        name: 'description',
+        message: 'Description for method:'
+      },
+      {
+        name: 'httpPath',
+        message: 'HTTP path:',
+        required: true
+      },
+      {
+        name: 'httpVerb',
+        message: 'HTTP verb:',
+        type: 'list',
+        choices: ['post', 'put', 'delete', 'get']
```

Comment:
> `choices: ['post', 'put', 'delete', 'get']`

Please add at least `head` and `patch`.

Here is a full list of methods as reported by `http.METHODS` in node@4:

```
> http.METHODS
[ 'CHECKOUT',
  'CONNECT',
  'COPY',
  'DELETE',
  'GET',
  'HEAD',
  'LOCK',
  'M-SEARCH',
  'MERGE',
  'MKACTIVITY',
  'MKCALENDAR',
  'MKCOL',
  'MOVE',
  'NOTIFY',
  'OPTIONS',
  'PATCH',
  'POST',
  'PROPFIND',
  'PROPPATCH',
  'PURGE',
  'PUT',
  'REPORT',
  'SEARCH',
  'SUBSCRIBE',
  'TRACE',
  'UNLOCK',
  'UNSUBSCRIBE' ]
```

It makes me wonder whether we should allow users to enter a custom method?

@superkhau @ritch @raymondfeng thoughts?